src/ssl.c

```diff
@@ -1595,6 +1592,9 @@ static int DupSSL(WOLFSSL* dup, WOLFSSL* ssl)
     dup->truncated_hmac = ssl->truncated_hmac;
 #endif
 
+    /* Restore rng option */
+    dup->options.weOwnRng = tmp_weOwnRng;
```

Comment:
This is not the right fix. After the memcpy of the options it should forcefully set the `dup->rng = NULL` and the `dup->options.weOwnRng = 0;` not touching anything else. We can't share the ownership of the RNG.

Reply:
The structures for `dup` and `ssl` are both initialized, and the each have their own RNG. This fix preserves the `weOwnRng` value, so that each one can be freed during `wolfSSL_free`. There is no sharing of the RNG.